src/importexport/musicxml/internal/musicxml/exportxml.cpp

```diff
@@ -7731,6 +7732,48 @@ static void writeStaffDetails(XmlWriter& xml, const Part* part)
     }
 }
 
+//---------------------------------------------------------
+//  writeInstrumentChange
+//---------------------------------------------------------
+
+/**
+ Write the instrument change.
+ */
+
+void ExportMusicXml::writeInstrumentChange(const InstrumentChange* instrChange)
+{
+    const Instrument* instr = instrChange->instrument();
+    const Part* part = instrChange->part();
+    const size_t partNr = muse::indexOf(m_score->parts(), part);
+    const int instNr = muse::value(m_instrMap, instr, -1);
+    static const std::wregex acc(L"[♭♯]");
+
+    m_xml.startElement("print");
+    m_xml.tag("part-name", instr->nameAsPlainText().replace(u"♭", u"b").replace(u"♯", u"#"));
+    if (instr->nameAsPlainText().contains(acc)) {
+        m_xml.startElement("part-name-display");
+        writeDisplayName(m_xml, instr->nameAsPlainText());
+        m_xml.endElement();
+    }
+    if (!instr->abbreviatureAsPlainText().isEmpty()) {
+        m_xml.tag("part-abbreviation", instr->abbreviatureAsPlainText().replace(u"♭", u"b").replace(u"♯", u"#"));
+        if (instr->abbreviatureAsPlainText().contains(acc)) {
+            m_xml.startElement("part-abbreviation-display");
+            writeDisplayName(m_xml, instr->abbreviatureAsPlainText());
+            m_xml.endElement();
+        }
+    }
+    m_xml.endElement();
+
+    writeInstrumentDetails(instrChange->instrument(), m_score->style().styleB(Sid::concertPitch));
```

Comment:
why not using `instr` here too?

Reply:
Thanks for your review! Actually found another thing that lead to invalid MusicXML output. Fixed that one, too.